library/java/src/io/envoyproxy/envoymobile/engine/JvmCallbackContext.java

```diff
@@ -10,118 +10,67 @@
 import io.envoyproxy.envoymobile.engine.types.EnvoyHTTPCallbacks;
 
 class JvmCallbackContext {
-  private enum FrameType {
-    NONE,
-    HEADERS,
-    METADATA,
-    TRAILERS,
-  }
-
+  private final JvmBridgeUtility bridgeUtility;
   private final EnvoyHTTPCallbacks callbacks;
 
-  // State-tracking for header accumulation
-  private Map<String, List<String>> headerAccumulator = null;
-  private FrameType pendingFrameType = FrameType.NONE;
-  private boolean pendingEndStream = false;
-  private long expectedHeaderLength = 0;
-  private long accumulatedHeaderLength = 0;
-
-  public JvmCallbackContext(EnvoyHTTPCallbacks callbacks) { this.callbacks = callbacks; }
+  public JvmCallbackContext(EnvoyHTTPCallbacks callbacks) {
+    bridgeUtility = new JvmBridgeUtility();
+    this.callbacks = callbacks;
+  }
 
   /**
    * Initializes state for accumulating header pairs via passHeaders, ultimately
    * to be dispatched via the callback.
    *
    * @param length,    the total number of headers included in this header block.
    * @param endStream, whether this header block is the final remote frame.
+   * @return Void,     not used for response callbacks.
    */
-  public void onHeaders(long length, boolean endStream) {
-    startAccumulation(FrameType.HEADERS, length, endStream);
+  public Void onHeaders(long headerCount, boolean endStream) {
+    assert bridgeUtility.validateCount(headerCount);
```

Comment:
This would crash if the validation returns false?

Reply:
Yes. It shouldn't ever fail, and the implementation is broken if it fails.